pkg/modelregistry/jsonvalues/convertjson.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jsonvalues
+
+import (
+	"fmt"
+	"github.com/onosproject/onos-config/pkg/modelregistry"
+	"github.com/onosproject/onos-config/pkg/store/change"
+	"strings"
+)
+
+func correctJSONPaths(jsonPathValues []*change.ConfigValue, roPaths modelregistry.ReadOnlyPathMap) ([]*change.ConfigValue, error) {
+	correctedPathValues := make([]*change.ConfigValue, 0)
+
+	for _, jsonPathValue := range jsonPathValues {
+		for ropath, subpaths := range roPaths {
+			if ropath == jsonPathValue.Path {
+				// Simple read-only path is leaf
+				if len(subpaths) != 1 {
+					return nil, fmt.Errorf("expected RO path %s to have only 1 subpath. Found %d", ropath, len(subpaths))
+				}
+				// FIXME - there is no support at present for setting the type when ReadOnly leaf is "config false"
+				correctedPathValues = append(correctedPathValues, jsonPathValue)
+				break
+			} else if strings.HasPrefix(jsonPathValue.Path, ropath) {
+				// Simple path, no lists
+				jsonSubPath := jsonPathValue.Path[len(ropath):]
+				for subPath, spType := range subpaths {
+					if jsonSubPath == subPath {
+						var newTypeValue *change.TypedValue
+						switch jsonPathValue.Type {
```

Comment:
should this switch statement handle also other types ?

Reply:
all other will be handled automatically in the default case - bool, string, enum etc. It's only because json marshaller changes all numeric values to float that I have to do this